IntegrationTests/Services/AWSS3IntegrationTests/S3SigV4ATests.swift

```diff
@@ -0,0 +1,207 @@
+        s3ControlClient = try S3ControlClient(region: region)
+        let createMRAPInput = CreateMultiRegionAccessPointInput(
+            accountId: accountId,
+            clientToken: UUID().uuidString.split(separator: "-").first!.lowercased(),
+            details: mrapConfig
+        )
+        _ = try await s3ControlClient.createMultiRegionAccessPoint(input: createMRAPInput).requestTokenARN
+
+        // Wait until MRAP creation finishes
+        var status: S3ControlClientTypes.MultiRegionAccessPointStatus? = .creating
+        repeat {
+            let seconds = 20.0
+            try await Task.sleep(nanoseconds: UInt64(seconds * Double(NSEC_PER_SEC)))
+
+            status = try await s3ControlClient.getMultiRegionAccessPoint(input: GetMultiRegionAccessPointInput(
+                accountId: accountId,
+                name: mrapName
+            )).accessPoint?.status
+        } while status == .creating
+
+        // Fetch MRAP alias then format & save MRAP ARN
+        // This MRAP ARN is used in place of bucket name in subsequent calls
+        mrapAlias = try await s3ControlClient.getMultiRegionAccessPoint(input: GetMultiRegionAccessPointInput(
+            accountId: accountId,
+            name: mrapName
+        )).accessPoint?.alias
+        mrapArn = String(format: mrapArnFormat, accountId, mrapAlias)
+    }
+
+    override func tearDown() async throws {
+        // MRAP must be deleted before deleting the linked bucket
+        try await deleteMRAP()
+        // Empty & delete the bucket
+        try await super.tearDown()
+    }
+
+    private func deleteMRAP() async throws {
+        // Delete the multi-region access point
+        _ = try await s3ControlClient.deleteMultiRegionAccessPoint(input: DeleteMultiRegionAccessPointInput(
+            accountId: accountId,
+            clientToken: UUID().uuidString.split(separator: "-").first!.lowercased(),
+            details: S3ControlClientTypes.DeleteMultiRegionAccessPointInput(name: mrapName)
+        ))
+
+        // Wait until MRAP has been deleted before returning
+        var mrapExists = true
+        repeat {
+            let seconds = 20.0
+            try await Task.sleep(nanoseconds: UInt64(seconds * Double(NSEC_PER_SEC)))
+
+            let mraps = try await s3ControlClient.listMultiRegionAccessPoints(input: ListMultiRegionAccessPointsInput(accountId: accountId)).accessPoints
+            mrapExists = checkMRAPExists(mraps ?? [])
+        } while mrapExists
+
+        // Wait some more before returning because
+        // deleting access point association with buckets takes more time
+        let seconds = 60.0
+        try await Task.sleep(nanoseconds: UInt64(seconds * Double(NSEC_PER_SEC)))
+    }
+
+    private func checkMRAPExists(_ mraps: [S3ControlClientTypes.MultiRegionAccessPointReport]) -> Bool {
+        for mrap in mraps {
+            if (mrap.name == mrapName) {
+                return true
+            }
+        }
+        return false
```

Comment:
All of this method can be replaced with:
```swift
return mraps.contains { $0.name == mrapName }
```

Reply:
Changed as suggested.